hummingbot/client/command/gateway_command.py

```diff
@@ -445,51 +442,167 @@ async def _prompt_for_wallet_address(
         wallet_address: str = response["address"]
         return wallet_address, additional_prompt_values
 
+    async def _get_balance_for_exchange(self, exchange_name: str):
+        gateway_connections = GatewayConnectionSetting.load()
+        gateway_instance = GatewayHttpClient.get_instance(self.client_config_map)
+        self.notify("Updating gateway balances, please wait...")
+        conf: Optional[Dict[str, str]] = GatewayConnectionSetting.get_connector_spec_from_market_name(
+            exchange_name)
+        if conf is None:
+            self.notify(
+                f"'{exchange_name}' is not available. You can add and review exchange with 'gateway connect'.")
+        else:
+            chain, network, address = (
+                conf["chain"], conf["network"], conf["wallet_address"]
+            )
+            native_token: str = native_tokens[chain]
+            tokens_str = conf.get("tokens", "")
+            tokens = [token.strip() for token in tokens_str.split(',')] if tokens_str else []
+            all_token = tokens + [native_token]
+
+            connector_chain_network = [
+                w for w in gateway_connections
+                if w["chain"] == chain and
+                w["network"] == network and
+                w["connector"] == conf["connector"] and
+                w["trading_type"] == conf["trading_type"]
+            ]
+
+            connector = connector_chain_network[0]['connector']
+            exchange = f"{connector}_{chain}_{network}"
+
+            try:
+                ex_bal = await gateway_instance.get_balances(
+                    chain, network, address, all_token, connector_chain_network[0]["connector"], fail_silently=True
+                )
+
+                allowance_resp = await gateway_instance.get_allowances(
+                    chain, network, address, tokens, connector_chain_network[0]["connector"], fail_silently=True
+                )
+
+                rows = []
+                for token in all_token:
+                    if ex_bal.get("balances") is None:
+                        bal = Decimal("0")
+                    else:
+                        bal = ex_bal["balances"].get(token, Decimal("0"))
+                    # Handle errors in allowance_responses_list
+                    if allowance_resp.get("approvals") is None:
+                        allowance = Decimal("0")
+                    else:
+                        allowance = allowance_resp["approvals"].get(token, Decimal("0"))
+
+                    rows.append({
+                        "Symbol": token.upper(),
+                        "Balance": PerformanceMetrics.smart_round(Decimal(str(bal)), 4),
+                        "Allowance": allowance,
```

Comment:
For display purposes, I suggest showing "999999+" if `allowance > 999999`. This should address @nikspz's comment.